spring-cloud-gcp-pubsub/src/test/java/com/google/cloud/spring/pubsub/support/DefaultSubscriberFactoryTests.java

```diff
@@ -144,6 +150,30 @@ public void testCreatePullRequest_nonNullMaxMessages() {
 		assertThat(request.getMaxMessages()).isEqualTo(Integer.MAX_VALUE);
 	}
 
+	@Test
+	public void testCreateSubscriber_validateSetProperties() {
+		GcpProjectIdProvider projectIdProvider = () -> "project";
+		DefaultSubscriberFactory factory = new DefaultSubscriberFactory(projectIdProvider, mockPubSubConfiguration);
+		factory.setCredentialsProvider(this.credentialsProvider);
+		PubSubConfiguration.Subscriber subscriber = new PubSubConfiguration.Subscriber();
+
+		subscriber.getFlowControl().setLimitExceededBehavior(FlowController.LimitExceededBehavior.Ignore);
+		subscriber.setMaxAckExtensionPeriod(2L);
+		subscriber.setParallelPullCount(2);
+		when(mockPubSubConfiguration.getSubscriber("defaultSubscription", "project"))
+				.thenReturn(subscriber);
+
+		Subscriber expectedSubscriber = factory.createSubscriber("defaultSubscription", (message, consumer) -> {
+		});
+
+		assertThat(expectedSubscriber.getFlowControlSettings().getLimitExceededBehavior())
+				.isEqualTo(FlowController.LimitExceededBehavior.Ignore);
+		assertThat(expectedSubscriber).hasFieldOrPropertyWithValue("maxAckExtensionPeriod", Duration.ofSeconds(2L))
+				.hasFieldOrPropertyWithValue("numPullers", 2);
+
+		factory.getThreadPoolTaskSchedulerMap().get("defaultSubscription").shutdown();
```

Comment:
Added tests to make sure that these properties are set correctly. However, I'm wondering if asserting directly on the field name and value will make the test brittle. Any thoughts?